syntax = "proto3";

package user;

service User {
  rpc GetUser(UserRequest) returns (UserResponse) {}
  rpc CreateUser(CreateUserRequest) returns (UserResponse) {}
  rpc DeleteUser(UserRequest) returns (UserResponse) {}

  rpc GetUsers(UsersRequest) returns (UsersResponse) {}
}

message UserRequest {
  string id = 1;
}

message CreateUserRequest { }

message UserResponse {
  string id = 1;
}

message UsersRequest {
  uint32 limit = 1;
  uint32 offset = 2;
}

message UsersResponse {
  repeated UserResponse users = 1;
}